pkg/controllers/vdb/httpscertrotation_reconciler.go

```diff
@@ -121,65 +127,117 @@ func (h *HTTPSCertRotationReconciler) Reconcile(ctx context.Context, _ *ctrl.Req
+		rotatehttpscerts.WithPollingCaCert(tlsData.caCert),
+		rotatehttpscerts.WithKey(secretName, keyConfig),
+		rotatehttpscerts.WithCert(secretName, certConfig),
+		rotatehttpscerts.WithCaCert(secretName, caCertConfig),
+		rotatehttpscerts.WithTLSMode(tlsData.tlsMode),
+		rotatehttpscerts.WithInitiator(tlsData.initiatorIP),
 	}
-	newCert := string(newSecret[corev1.TLSCertKey])
-	currentCert := string(currentSecret[corev1.TLSCertKey])
-	rotated, err := security.VerifyCert(initiatorPod.GetPodIP(), builder.VerticaHTTPPort, newCert, currentCert, h.Log)
+	h.VRec.Eventf(h.Vdb, corev1.EventTypeNormal, events.HTTPSCertRotationStarted,
+		"Starting https cert rotation with secret name %s and mode %s",
+		h.Vdb.Spec.HTTPSTLSSecret, tlsData.tlsMode)
+	err := h.Dispatcher.RotateHTTPSCerts(ctx, opts...)
 	if err != nil {
-		h.Log.Error(err, "https cert rotation aborted. Failed to verify new https cert for "+
-			initiatorPod.GetPodIP())
-		return ctrl.Result{}, err
+		h.VRec.Eventf(h.Vdb, corev1.EventTypeWarning, events.HTTPSCertRotationFailed,
+			"Failed to rotate https cert with secret name %s and mode %s", h.Vdb.Spec.HTTPSTLSSecret, tlsData.tlsMode)
+		return err
 	}
-	if rotated == 2 {
-		h.Log.Info("https cert rotation aborted. Neither new nor current https cert is in use")
-		return ctrl.Result{Requeue: true}, nil
-	}
-	if rotated == 0 {
-		h.Log.Info("https cert rotation skipped. new https cert is already in use on " + initiatorPod.GetPodIP())
-	} else {
-		currentSecretName := h.Vdb.GetHTTPSTLSSecretNameInUse()
-		h.Log.Info("ready to rotate certi from " + currentSecretName + " to " + h.Vdb.Spec.HTTPSTLSSecret)
-		var keyConfig, certConfig, caCertConfig, secretName string
-		switch {
-		case secrets.IsAWSSecretsManagerSecret(h.Vdb.Spec.HTTPSTLSSecret):
-			keyConfig, certConfig, caCertConfig = GetAWSCertsConfig(h.Vdb)
-			secretName = secrets.RemovePathReference(h.Vdb.Spec.HTTPSTLSSecret)
-		default:
-			keyConfig, certConfig, caCertConfig = GetK8sCertsConfig(h.Vdb)
-			secretName = h.Vdb.Spec.HTTPSTLSSecret
-		}
-		opts := []rotatehttpscerts.Option{
-			rotatehttpscerts.WithPollingKey(string(newSecret[corev1.TLSPrivateKeyKey])),
-			rotatehttpscerts.WithPollingCert(newCert),
-			rotatehttpscerts.WithPollingCaCert(string(newSecret[corev1.ServiceAccountRootCAKey])),
-			rotatehttpscerts.WithKey(secretName, keyConfig),
-			rotatehttpscerts.WithCert(secretName, certConfig),
-			rotatehttpscerts.WithCaCert(secretName, caCertConfig),
-			rotatehttpscerts.WithTLSMode(h.Vdb.Spec.HTTPSTLSMode),
-			rotatehttpscerts.WithInitiator(initiatorPod.GetPodIP()),
+	h.VRec.Eventf(h.Vdb, corev1.EventTypeNormal, events.HTTPSCertRotationSucceeded,
+		"Successfully rotated https cert with secret name %s and mode %s", h.Vdb.Spec.HTTPSTLSSecret, tlsData.tlsMode)
+
+	return err
+}
+
+// buildHTTPSTLSUpdateData constructs the data needed for tls update
+func (h *HTTPSCertRotationReconciler) buildHTTPSTLSUpdateData(ctx context.Context,
```

Comment:
collect tls info for vclusterops in a single place

Reply:
address the comments?